src/v/cluster/cluster_utils.cc

```diff
@@ -185,6 +185,21 @@ std::vector<topic_result> create_topic_results(
     return results;
 }
 
+std::vector<topic_result> create_topic_results(
+  const std::vector<custom_assignable_topic_configuration>& requests,
+  errc error_code) {
+    std::vector<topic_result> results;
+    results.reserve(requests.size());
+    std::transform(
+      std::cbegin(requests),
+      std::cend(requests),
```

Comment:
nit: for non-templated code (generic over container type) I'm not sure that std::begin offers any advantage